src/main/java/Storage.java

```diff
@@ -0,0 +1,162 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Stores the list of tasks, and is responsible for
+ * loading and changing the list of tasks in the hard disk.
+ */
+public class Storage {
+    private final String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the list of tasks from the hard disk into Duke.
+     * @return An ArrayList containing all the tasks saved in hard disk
+     * @throws DukeException if the file cannot be found in the hard disk
+     */
+    public ArrayList<Task> load() throws DukeException {
+        String currDirectory = System.getProperty("user.dir");
+        java.nio.file.Path path = java.nio.file.Paths.get(currDirectory, "data");
+
+        ArrayList<Task> taskList = new ArrayList<>();
+        File tasksFile = new File(filePath);
+        try {
+            if (!java.nio.file.Files.exists(path)) {
+                java.nio.file.Files.createDirectory(path);
+            }
+            if (!tasksFile.exists()) {
+                tasksFile.createNewFile();
+            }
+        } catch (IOException e) {
+            throw new DukeException("Unable to create file.", e);
+        }
+
+        try {
+            Scanner readFile = new Scanner(tasksFile);
+
+            while (readFile.hasNextLine()) {
+                String taskName;
+                String nextTask = readFile.nextLine();
+                boolean done = false;
+                if (nextTask.charAt(4) == '1') {
+                    done = true;
+                }
+                if (nextTask.startsWith("T")) {
+                    taskName = nextTask.substring(8);
+                    taskList.add(new Todo(taskName, done));
+                } else {
+                    int index = nextTask.lastIndexOf("|");
+                    taskName = nextTask.substring(8, index - 1);
+                    String dateTime = nextTask.substring(index + 2);
+                    String dateString = dateTime.substring(0, 11);
+                    DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
+                    LocalDate date = LocalDate.parse(dateString, dateFormatter);
+                    String time = dateTime.substring(12);
+
+                    if (nextTask.startsWith("E")) {
+                        taskList.add(new Event(taskName, done, date, time));
+                    } else if (nextTask.startsWith("D")) {
+                        taskList.add(new Deadline(taskName, done, date, time));
+                    }
+                }
+
```

Comment:
I appreciate the blank link between the variable assignment and subsequent logic.